lib/ansible/modules/cloud/vagrant/vagrant_box.py

```diff
@@ -0,0 +1,290 @@
+
+        opts_list = []
+        if os.path.exists(self.box):
+            if self.name:
+                box = self.name
+            else:
+                box = os.path.basename(self.box)
+        else:
+            box = self.box
+        if self.force:
+            opts_list.append('--force')
+        if self.provider:
+            opts_list.append('--provider %s' % self.provider)
+        if self.version is not "latest":
+            opts_list.append('--box-version "%s"' % self.version)
+
+        cmd = "%s box remove %s %s" % (self.module.get_bin_path('vagrant', required=True), box, " ".join(opts_list))
+        rc, out, err = self.module.run_command(cmd)
+
+        if rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        elif "interface with the UI" in err:
+            self.module.fail_json(msg="The box: '%s' is still in use by at least one Vagrant environment, use 'force' flag to force the deletion" % self.box)
+        elif "could not be found" in err:
+            self.module.exit_json(changed=False, stdout="The box: '%s' version: '%s' provider: '%s' already removed" % (box, self.version, self.provider))
+        else:
+            self.module.fail_json(msg=err, rc=rc)
+
+    def update(self):
+        if not self.ispresent() and self.version is not "latest":
+            self.module.fail_json(msg="The box: '%s' version: '%s' provider: '%s' does not exist" % (self.box, self.version, self.provider))
+
+        if os.path.exists(self.box):
+            box = os.path.basename(self.box)
+        else:
+            box = self.box
+
+        cmd = "%s box update --box %s --provider %s" % (self.module.get_bin_path('vagrant', required=True), box, self.provider)
+        rc, out, err = self.module.run_command(cmd)
+
+        if "is running the latest version" in out:
+            self.module.exit_json(changed=False, rc=rc, stdout=out, stderr=err)
+        elif rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        else:
+            self.module.fail_json(msg=err)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            box=dict(required=True, aliases=['path', 'url']),
+            state=dict(choices=['present', 'absent', 'updated'], default="present"),
+            force=dict(default=False, type='bool'),
+            version=dict(default="latest"),
+            provider=dict(choices=['hyperv', 'libvirt', 'virtualbox', 'vmware_desktop'], default="virtualbox"),
+            insecure=dict(default=False, type='bool'),
+            trust_location=dict(default=False, type='bool'),
+            cacert=dict(type='path'),
+            capath=dict(type='path'),
+            cert=dict(type='path'),
+            name=dict(),
+            checksum=dict(),
+            checksum_type=dict(choices=['md5', 'sha1', 'sha256'], default="sha256"),
+        ),
+        supports_check_mode=False,
+    )
+
```

Comment:
add `module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')` to ensure all commands have an expected LANG